src/components/KMultiselect/KMultiselect.vue

```diff
@@ -0,0 +1,911 @@
+      // This prevents the popover from displaying "detached" from the KSelect
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      if (popper.value && typeof popper.value.updatePopper === 'function') {
+        nextTick(() => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          popper.value.updatePopper()
+        })
+      }
+    }, { deep: true, immediate: true })
+
+    return {
+      // keys and ids
+      key,
+      stagingKey,
+      multiselectId,
+      multiselectInputId,
+      multiselectTextId,
+      multiselectSelectedItemsStagingId,
+      multiselectSelectedItemsId,
+      // text
+      getPlaceholderText,
+      filterStr,
+      // items
+      unfilteredItems,
+      sortedItems,
+      selectedItems,
+      invisibleSelectedItemsStaging,
+      visibleSelectedItemsStaging,
+      invisibleSelectedItems,
+      visibleSelectedItems,
+      hiddenItemsTooltip,
+      // style and attributes
+      widthStyle,
+      inputWidth,
+      modifiedAttrs,
+      isHovered,
+      isFocused,
+      isDisabled,
+      isReadonly,
+      popper,
+      boundKPopAttributes,
+      // functions
+      sortItems,
+      handleItemSelect,
+      clearSelection,
+      triggerFocus,
+      onQueryChange,
+      onInputFocus,
+      onPopoverOpen,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+.k-multiselect {
+  width: fit-content; // necessary for correct placement of popup
+
+  // off screen area for checking selections before display
+  .staging-area {
+    position: absolute;
+    left: -99999px;
+  }
+
+  .k-multiselect-selections {
+    --KBadgeMaxWidth: 100px;
+    -webkit-box-sizing: border-box;
+    -moz-box-sizing: border-box;
+    box-sizing: border-box;
+    padding-left: 16px;
+    padding-right: 23px;
+
+    &.staging {
+      -webkit-box-sizing: border-box;
+      -moz-box-sizing: border-box;
+      box-sizing: border-box;
+      position: relative;
+      height: auto;
+      padding-left: 16px;
+      padding-right: 23px;
+    }
+
+    .hidden-selection-count {
+      // match dismissable height
+      --KBadgeLineHeight: 21px;
+
+      &.hidden {
+        visibility: hidden;
+      }
+    }
+  }
+
+  .k-multiselect-icon {
+    position: absolute;
+    z-index: 1;
+    right: 1px;
+    top: 1px;
+
+    .k-multiselect-chevron-icon {
+      position: relative;
+      top: 11px;
+      right: 10px;
+    }
+
+    .k-multiselect-clear-icon {
+      position: relative;
+      top: 8px;
+      right: 10px;
+    }
+  }
+
+  .k-multiselect-trigger {
+    display: inline-block;
+    position: relative;
+    // mimic input's box shadow styling
+    background-color: var(--KInputBackground, var(--white));
+    box-shadow: inset 0 0 0 1px var(--KInputBorder, var(--grey-300)) !important;
```

Comment:
I think you could use the existing mixin for this?

https://github.com/Kong/kongponents/blob/6afe443a3c8e083a38cd7dc6fee753929c2d9783/src/styles/_mixins.scss#L22-L26